ng2-components/ng2-alfresco-upload/src/services/file-uploading.service.ts

```diff
@@ -0,0 +1,30 @@
+/*!
+ * @license
+ * Copyright 2016 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Rx';
+
+@Injectable()
+export class FileUploadService {
+    private remove = new Subject<string>();
+    public onRemoveFile = this.remove.asObservable();
```

Comment:
You can make "remove" member public. The Subject is already an Observable inside, so anyone can emit the event just via "service.remove.next(obj)", and subscribe to the changes. See examples here: https://github.com/Alfresco/alfresco-ng2-components/blob/development/ng2-components/ng2-alfresco-core/src/services/upload.service.ts#L37